api/v1alpha1/spinapp_types.go

```diff
@@ -80,6 +80,9 @@ type SpinAppSpec struct {
 
 	// Resources defines the resource requirements for this app.
 	Resources Resources `json:"resources,omitempty"`
+
+	// Otel provides Kubernetes Bindings to Otel Variables.
+	Otel []OtelVar `json:"otel,omitempty"`
```

Comment:
Rather than being a bunch of loose kvs in a list I'd much rather have this be less flexible and more structured

Like:

type OtelConfig struct {
 Exporter string
}

Reply:
So like this: 
```
  otel:
    - OTEL_EXPORTER_OTLP_ENDPOINT: http://otel-collector-opentelemetry-collector.default.svc.cluster.local:4318
```

rather than this: 
```
  otel:
    - name: OTEL_EXPORTER_OTLP_ENDPOINT
      value: http://otel-collector-opentelemetry-collector.default.svc.cluster.local:4318
```

Reply:
no like this:

```yaml
otel:
  endpoint: http://otel-collector-opentelemetry-collector.default.svc.cluster.local:4318
```

that we convert into envs/however the shim needs config in the operator

Reply:
More specifically I prefer the bag of kv's approach b/c than we don't have to churn on the CRD, we can churn on the validation in the operator. OTEL has a __lot__ of different configuration options and I really don't want to bake them all into our CRD.

Reply:
Building a stable interface is much cheaper (long term) than adding loosely defined values specific to a single implementation, and our entire design thus far has been to avoid the latter.

Reply:
I want to push back on two things:

1. I expect that there will be very little difference in the ENV keys between executors. The OTEL env vars are standardized for a reason. In the case where an executor does need some very specific env vars it would probably be in its own field outside of the otel field.
2. The loose KV approach is still defining a stable interface. The stable interface just lives at the validation layer in the operator where we validate which keys are allowed.

Do they still have to uninstall/reinstall CRDs when fields are added?

Reply:
> Do they still have to uninstall/reinstall CRDs when fields are added?
Nope

2) The downside there is that it's not a well known, published interface - so offline validation tools and editor autocomplete can never work properly - with a well defined API that comes with an OpenAPI spec, it can.

Reply:
> The downside there is that it's not a well known, published interface - so offline validation tools and editor autocomplete can never work properly - with a well defined API that comes with an OpenAPI spec, it can.

That is a downside.

> Do they still have to uninstall/reinstall CRDs when fields are added?
> Nope

That changes my mind. How does it work then, do they just apply the new CRD and it patches?

Reply:
Yep!